packages/salesforcedx-vscode-soql/test/vscode-integration/queryDataView/queryDataFileService.test.ts

```diff
@@ -8,62 +8,104 @@
 import { expect } from 'chai';
 import * as fs from 'fs';
 import * as path from 'path';
+import * as sinon from 'sinon';
 import * as vscode from 'vscode';
-import { QUERY_RESULTS_DIR_PATH } from '../../../src/constants';
 import {
   CsvDataProvider,
   JsonDataProvider
 } from '../../../src/queryDataView/dataProviders';
 import { FileFormat } from '../../../src/queryDataView/queryDataFileService';
-import { mockQueryData, TestFileService } from '../testUtilities';
+import {
+  mockQueryData,
+  MockTextDocumentProvider,
+  TestFileService
+} from '../testUtilities';
+
+export const QUERY_RESULTS_DIR_PATH = path.join(
+  'scripts',
+  'soql',
+  'query-results'
+);
 
 describe('Query Data File Service', () => {
+  let mockTextDocument: vscode.TextDocument;
+  let docProviderDisposable: vscode.Disposable;
   const documentName = 'example.soql';
   const workspacePath = vscode.workspace.workspaceFolders![0].uri.fsPath;
   const testResultsDirPath = path.join(workspacePath, QUERY_RESULTS_DIR_PATH);
+  let sandbox: sinon.SinonSandbox;
+
+  function createResultsDirectory() {
+    fs.mkdirSync(testResultsDirPath, {
+      recursive: true
+    });
+  }
+
+  beforeEach(async () => {
+    sandbox = sinon.createSandbox();
+    docProviderDisposable = vscode.workspace.registerTextDocumentContentProvider(
+      'sfdc-test',
+      new MockTextDocumentProvider()
+    );
+    mockTextDocument = await vscode.workspace.openTextDocument(
+      vscode.Uri.parse('sfdc-test:test/examples/soql/mocksoql.soql')
+    );
+    createResultsDirectory();
+  });
 
   afterEach(() => {
     // delete the query-results directory and its files.
     // @ts-ignore
     fs.rmdirSync(testResultsDirPath, { recursive: true });
+    sandbox.restore();
   });
 
   it('should use the correct data provider', () => {
     const csvFileService = new TestFileService(
       mockQueryData,
       FileFormat.CSV,
-      documentName
+      mockTextDocument
     );
     expect(csvFileService.getDataProvider()).instanceOf(CsvDataProvider);
 
     const jsonFileService = new TestFileService(
       mockQueryData,
       FileFormat.JSON,
-      documentName
+      mockTextDocument
     );
     expect(jsonFileService.getDataProvider()).instanceOf(JsonDataProvider);
   });
 
-  it('will save json file to disk on save', () => {
+  it('will save json file to disk on save', async () => {
     const jsonFileService = new TestFileService(
       mockQueryData,
       FileFormat.JSON,
-      documentName
+      mockTextDocument
     );
 
-    const savedFilePath = jsonFileService.save();
+    const mockURI = {
+      path: `${testResultsDirPath}/${documentName}`
+    } as vscode.Uri;
+    sandbox.stub(vscode.window, 'showSaveDialog').resolves(mockURI);
```

Comment:
Doesn't matter in this specific case, but when building paths for tests it's a good idea to use `path.join` to avoid any windows issues while testing.

Reply:
thanks, see the `fix: clean up tests, use path.join` commit.